include/kernel.h

```diff
@@ -4540,7 +4540,23 @@ extern void z_mem_pool_free_id(struct k_mem_block_id *id);
  */
 
 /**
- * @brief Allocate memory from heap.
+ * @brief Allocate memory from the heap with a specified alignment.
+ *
+ * This routine provides semantics similar to aligned_alloc(3); memory is
```

Comment:
My undertsanding of `(3)` here is that it UNIX man page style. But consider that manual pages are tied to some implementation, may be different on different systems and may change. It is better to reference the related standard instead if needed.

Reply:
I removed the `3` and added the following:
```
C11 standard (ISO/IEC 9899:2011):
7.22.3.1 The aligned_alloc function (p: 347-348)
```